### PR TITLE
RAC-395: Validation error with direct link to job only on internal API

### PR DESCRIPTION
**Pull request by StevenVAIDIE**
**Description (for Contributor and Core Developer)**
In this PR, change the validation error to give the direct link to job only on internal API.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Comment by tseho** [comment]
There is no context like `internal_api` in the constraint ?
It's something we do for the normalizers https://github.com/akeneo/pim-community-dev/pull/13502/files#diff-2327788c38e025215f97d7fb95d168aa9b5f59e8024d8204a20f2732079b75cdR19
The constraint would then return only one message, and the correct one by being aware of the context.
You wouldn't have to change the normalizer.

File: src/Akeneo/Pim/Enrichment/Component/Product/Normalizer/InternalApi/ConstraintViolationNormalizer.php [hunk 1]
```diff
@@ -93,4 +95,17 @@ protected function getStandardPath(ConstraintViolation $violation)
 
         return Inflector::tableize($violation->getPropertyPath());
     }
+
+    private function getInternalApiMessage(ConstraintViolation $violation): string
+    {
+        if ( ! $violation->getConstraint()) {
```
Comment (ValentinMumble) [inline src/Akeneo/Pim/Enrichment/Component/Product/Normalizer/InternalApi/ConstraintViolationNormalizer.php:101]:
```suggestion
        if (!$violation->getConstraint()) {
```

File: src/Akeneo/Pim/Structure/Component/Validator/Constraints/BlacklistedAttributeCodeValidator.php [hunk 2]
```diff
@@ -31,8 +46,22 @@ public function validate($attributeCode, Constraint $constraint): void
...
     }
+
+    private function addInternalViolation(string $attributeCode, BlacklistedAttributeCode $constraint): void
+    {
+        $jobExecutionId = $this->getBlacklistedAttributeJobExecutionId->forAttributeCode($attributeCode);
+        $internalApiMessage = $this->translator->trans(
+            $constraint->internalAPIMessage,
+            ['{{ link }}' => "#" . $this->router->generate(self::JOB_TRACKER_ROUTE, ['id' => $jobExecutionId])],
+            'validators'
+        );
```
Comment (ValentinMumble) [inline src/Akeneo/Pim/Structure/Component/Validator/Constraints/BlacklistedAttributeCodeValidator.php:63]:
```suggestion
        $jobExecutionLink = sprintf('#%s', $this->router->generate(self::JOB_TRACKER_ROUTE, ['id' => $jobExecutionId]));
        $internalApiMessage = $this->translator->trans(
            $constraint->internalAPIMessage,
            ['{{ link }}' => $jobExecutionLink],
            'validators'
        );
```

**Review by ValentinMumble — APPROVED** [review]

**Review by juliensnz — APPROVED** [review]

**Comment by StevenVAIDIE** [comment]
> There is no context like `internal_api` in the constraint ?
> It's something we do for the normalizers https://github.com/akeneo/pim-community-dev/pull/13502/files#diff-2327788c38e025215f97d7fb95d168aa9b5f59e8024d8204a20f2732079b75cdR19
> The constraint would then return only one message, and the correct one by being aware of the context.
> You wouldn't have to change the normalizer.

Unfortunately, it's not possible. The only thing that we can pass through the validator is groups. We cannot use them here because Attribute class have GroupSequenceProvider the order of group given by Controller are not keeped and the group in the context on Validator is always Attribute.

The right way is to don't translated message and keep the front doing the translation. But, it requires reworking the front to deal with this (common field on UiBundle)

**Pull request merged**